client/js/widgets/button.js

```diff
@@ -53,12 +53,19 @@ export class Button extends Widget {
 
     if(this.p('debug'))
       $('#debugButtonOutput').textContent = '';
-
+   let thisID = this.p('id');
     const variables = {
       playerName,
-      playerColor
+      playerColor,
+      "thisID" : thisID
     };
-    const collections = {};
+    const thisButton = (Array.from(widgets.values())).filter(function(w) {
+     if(w.p("id")==thisID)
+        return true;
+    });
```

Comment:
You obviously copied this from the other code that creates a collection. Which is a reasonable thing to do.

The parenthesis around `Array.from` are not necessary.

`if(w.p("id")==thisID) return true;` could just be `return w.p("id")==thisID;`. The whole `function(w) { return w.p("id")==thisID; }` can also be written as `w=>w.p("id")==thisID`. You don't even need to look at the ID though. A `w==this` would be fine.

In the end you're doing all of that to go through all the `widgets` and only return the one you're currently handling. So in the end you could just write `const thisButton = [ this ];` because you already know exactly what should be in the array.

I'm guessing you weren't sure if the `widgets` array contains the same stuff as the `this`. It does.

Reply:
You've hit the nail on the head, I didn't realise they were exactly the same